client/src/observatory/results.tsx

```diff
@@ -0,0 +1,668 @@
+    if (firstUrl.hostname === lastUrl.hostname) {
+      return host;
+    }
+    return `${firstUrl.hostname} → ${lastUrl.hostname}`;
+  } catch (e) {
+    return host;
+  }
+}
+
+function CountdownButton({
+  host,
+  from,
+  duration,
+  title,
+  onClickHandler,
+}: {
+  host: string;
+  from: string;
+  duration: number;
+  title: string;
+  onClickHandler: Function;
+}) {
+  function calculateRemainingTime() {
+    const endTime = new Date(from).getTime() + duration * 1000;
+    return Math.max(0, endTime - new Date().getTime());
+  }
+  const [remainingTime, setRemainingTime] = useState(calculateRemainingTime());
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setRemainingTime(calculateRemainingTime());
+    }, 1000);
+
+    return () => clearInterval(interval);
+  });
+
+  const gleanClick = useGleanClick();
+
+  function rescan() {
+    gleanClick(`${OBSERVATORY}: click: rescan`);
+    onClickHandler();
+  }
+
+  const isExpired = remainingTime <= 0;
+  const remainingSecs = Math.floor(remainingTime / 1000) + 1;
+  const progressPercent = (remainingSecs * 100) / 60;
+  return !isExpired ? (
+    <Button isDisabled={true}>
+      <div
+        className="progress"
+        style={{
+          background: `conic-gradient(var(--button-color) 0grad, ${progressPercent}%, rgba(0,0,0,0) ${progressPercent}% 100%)`,
+        }}
+      ></div>
+      <small>
+        Wait {remainingSecs}s to {title.toLowerCase()}
+      </small>
+    </Button>
+  ) : (
+    <Button onClickHandler={rescan}>{title}</Button>
+  );
+}
+
+function ObservatoryTests({ result }: { result: ObservatoryResult }) {
+  return Object.keys(result.tests).length !== 0 ? (
+    <section className="tab-content">
+      <figure className="scroll-container">
+        <table className="tests">
+          <thead>
+            <tr>
+              <th>Test</th>
+              <th>Score</th>
+              <th>Reason</th>
+              <th>Recommendation</th>
+            </tr>
+          </thead>
+          <tbody>
+            {Object.entries(result.tests).map(([name, test]) => {
+              return (
+                <tr key={name}>
+                  <td data-header="Test">
+                    <Link href={test.link}>{test.title}</Link>
+                  </td>
+                  {test.pass === null ? (
+                    <td data-header="Score">-</td>
+                  ) : (
+                    <td className="score" data-header="Score">
+                      <span>
+                        <span className="obs-score-value">
+                          {fixMinusSymbol(`${test.score_modifier}`)}
+                        </span>
+                        <PassIcon pass={test.pass} />
+                      </span>
+                    </td>
+                  )}
+                  <td
+                    data-header="Reason"
+                    dangerouslySetInnerHTML={{
+                      __html: test.score_description,
+                    }}
```

Comment:
We might want to consider returning Markdown instead, and rendering it here?

Reply:
Discussed this with Florian, and we settled on HTML. This would need a change in the backend as well, so I'd prefer leaving it as it is.